.github/workflows/process_new_pr.yml

```diff
@@ -0,0 +1,127 @@
+          echo "pr_number=${{ github.event.pull_request.number }}" >> $GITHUB_OUTPUT
+      - name: Check if PR Author is a New Contributor
+        id: check-new-contributor
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const author = '${{ github.event.pull_request.user.login }}';
+            const owner = '${{ github.event.repository.owner.login }}';
+            const repo = '${{ github.event.repository.name }}';
+            
+            try {
+              const { data } = await github.rest.search.issuesAndPullRequests({
+                q: `repo:${owner}/${repo} type:pr is:merged author:${author}`,
+              });
+              return data.total_count === 0;
+            } catch (error) {
+              console.error('Error checking contributor status:', error);
+              return false;
+            }
+      - name: Check for Assigned Reviewers and Teams
+        id: check-reviewers
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const pr = context.payload.pull_request;
+            const hasReviewers = pr.requested_reviewers?.length > 0;
+            const hasTeams = pr.requested_teams?.length > 0;
+            core.setOutput('has_reviewers', hasReviewers);
+            core.setOutput('has_teams', hasTeams);
+  manage_reviewers_and_comments:
+    name: Assign Reviewers and Post Comments
+    needs: check_contributor_and_reviewers
+    runs-on: ubuntu-22.04
+    permissions:
+      pull-requests: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Merge develop and set up dependencies
+        uses: ./.github/actions/merge-develop-and-setup-python
+
+      - name: Assign PR Reviewers as Assignees
+        if: needs.check_contributor_and_reviewers.outputs.has_reviewers == 'true'
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const owner = '${{ needs.check_contributor_and_reviewers.outputs.owner }}';
+            const repo = '${{ needs.check_contributor_and_reviewers.outputs.repo }}';
+            const pr_number = parseInt('${{ needs.check_contributor_and_reviewers.outputs.pr_number }}');
+            
+            try {
+              const { data: pr } = await github.rest.pulls.get({
+                owner,
+                repo,
+                pull_number: pr_number
+              });
+              
+              const reviewers = pr.requested_reviewers.map(r => r.login);
+              
+              if (reviewers.length > 0) {
+                await github.rest.issues.addAssignees({
+                  owner,
+                  repo,
+                  issue_number: pr_number,
+                  assignees: reviewers
```

Comment:
don't we only want to assign the first-pass reviewer at first? or is that policy changing?